src/wp-includes/canonical.php

```diff
@@ -63,10 +63,22 @@ function redirect_canonical( $requested_url = null, $do_redirect = true ) {
 		return;
 	}
 
-	if ( ! $requested_url && isset( $_SERVER['HTTP_HOST'] ) ) {
+	// www.example.com vs. example.com
+	$user_home = parse_url( home_url( '/' ) );
+
+	if ( ! $requested_url ) {
 		// Build the URL in the address bar.
-		$requested_url  = is_ssl() ? 'https://' : 'http://';
-		$requested_url .= $_SERVER['HTTP_HOST'];
+		$requested_url = is_ssl() ? 'https://' : 'http://';
+		if ( isset( $_SERVER['HTTP_HOST'] ) ) {
+			$requested_url .= $_SERVER['HTTP_HOST'];
+		} else {
+			// Fallback for http1.0 where requested host and port are unknown
+			// In this case correct redirect no-www <=> yes-www is not possible
```

Comment:
```suggestion
			/*
			 * Fallback for HTTP/1.0 where the requested host and port are unknown.
			 * In this case, correctly redirecting no-www <=> yes-www is not possible.
			 */
```
Just a couple of adjustments to use the multiline comment format per the [inline documentation standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments), add periods and some minor rephrasing. 🙂